lib/hanami/slice.rb

```diff
@@ -981,6 +987,11 @@ def load_router(inspector:)
         end
       end
 
+      def render_errors?
+        config.render_errors ||
+          (config.render_detailed_errors && Hanami.bundled?("hanami-webconsole"))
```

Comment:
@timriley Why do we need `hanami-webconsole` knowledge here?

Isn't webconsole a Rack middleware that is able to catch exceptions and present them in a nice page?

Reply:
Yes, it is, but if it’s not bundled, then the value of render_detailed_errors here is moot: the middleware won’t be activated, it won’t catch the exceptions raised from our procs below, and your not found pages will come back as 500 errors instead of 404 responses.

This method is just mirroring the logic above from inside load_router. The duplication isn’t ideal but it seems manageable for now.

Reply:
But why our 404 responses are raising exceptions?
Given the router doesn't raise any exceptions.

Reply:
I might be able to explore some other options here where we have our standard RenderErrors middleware provide some sensible fallbacks in the case of render_errors setting being false, but this here is the simplest fix for the short term, and it won't prevent us from taking a more sophisticated approach later.

Reply:
@jodosha the 404s are being made into exceptions so our error rendering features (render_errors or rendered_detailed_errors) can pick them up and handle them as appropriate. 

I don't think you spent much time familiarizing yourself that work because it happened during one of those phases when you got very busy with non-hanami matters, which is why this might seem foreign to you now. 

See:

- https://github.com/hanami/hanami/pull/1309
- https://github.com/hanami/hanami/pull/1311

Reply:
@timriley I'm fine with the short-term fix, but we need to discuss simplifying the implementation.

To summarize: the router doesn't raise exceptions, but we raise it to print a nice error, and now we're handling those exceptions to not bubble up. 😉